types/validator_set.go

```diff
@@ -653,253 +652,25 @@ func (vals *ValidatorSet) UpdateWithChangeSet(changes []*Validator) error {
-				return nil
-			}
-		}
-	} else {
-		talliedVotingPower, err = verifyCommitLightSingle(
-			chainID, vals, commit, votingPowerNeeded, cacheSignBytes)
-		if err != nil {
-			return err
-		} else if talliedVotingPower > votingPowerNeeded {
-			return nil
-		}
-	}
-	return ErrNotEnoughVotingPowerSigned{Got: talliedVotingPower, Needed: votingPowerNeeded}
+	return VerifyCommitLight(chainID, vals, blockID, height, commit)
 }
 
 // VerifyCommitLightTrusting verifies that trustLevel of the validator set signed
 // this commit.
-//
-// NOTE the given validators do not necessarily correspond to the validator set
-// for this commit, but there may be some intersection.
-//
-// This method is primarily used by the light client and does not check all the
-// signatures.
 func (vals *ValidatorSet) VerifyCommitLightTrusting(chainID string, commit *Commit, trustLevel tmmath.Fraction) error {
-	// sanity checks
-	if trustLevel.Denominator == 0 {
-		return errors.New("trustLevel has zero Denominator")
-	}
-	if commit == nil {
-		return errors.New("nil commit")
-	}
-
-	var (
-		talliedVotingPower int64
-		seenVals           = make(map[int32]int, len(commit.Signatures)) // validator index -> commit index
-		err                error
-		cacheSignBytes     = make(map[string][]byte, len(commit.Signatures))
-	)
-
-	// Safely calculate voting power needed.
-	totalVotingPowerMulByNumerator, overflow := safeMul(vals.TotalVotingPower(), int64(trustLevel.Numerator))
-	if overflow {
-		return errors.New("int64 overflow while calculating voting power needed. please provide smaller trustLevel numerator")
-	}
-	votingPowerNeeded := totalVotingPowerMulByNumerator / int64(trustLevel.Denominator)
-
-	bv, ok := batch.CreateBatchVerifier(vals.GetProposer().PubKey)
-	if ok && len(commit.Signatures) > 1 {
-		for idx, commitSig := range commit.Signatures {
-			// No need to verify absent or nil votes.
-			if !commitSig.ForBlock() {
-				continue
-			}
-
-			// We don't know the validators that committed this block, so we have to
-			// check for each vote if its validator is already known.
-			valIdx, val := vals.GetByAddress(commitSig.ValidatorAddress)
-
-			if val != nil {
-				// check for double vote of validator on the same commit
-				if firstIndex, ok := seenVals[valIdx]; ok {
-					secondIndex := idx
-					return fmt.Errorf("double vote from %v (%d and %d)", val, firstIndex, secondIndex)
-				}
-				seenVals[valIdx] = idx
-
-				// Validate signature.
-				voteSignBytes := commit.VoteSignBytes(chainID, int32(idx))
-				// cache the signed bytes in case we fail verification
-				cacheSignBytes[string(val.PubKey.Bytes())] = voteSignBytes
-				// if batch verification is supported add the key, sig and message to the verifier
-				if err := bv.Add(val.PubKey, voteSignBytes, commitSig.Signature); err != nil {
-					return err
-				}
-
-				talliedVotingPower += val.VotingPower
-
-				if talliedVotingPower > votingPowerNeeded {
-					return nil
```

Comment:
I've realised it might not be so clear in terms of validation what we are changing. Here we return before actually running `bv.Verify()`. This should instead `break`